purchasing/conductor/views.py

```diff
@@ -267,3 +271,32 @@ def assign(contract_id, flow_id, user_id):
     db.session.commit()
     flash('Successfully assigned to {}!'.format(user.email), 'alert-success')
     return redirect(url_for('conductor.index'))
+
+@blueprint.route('/upload_new', methods=['GET', 'POST'])
+def upload():
+    form = FileUpload()
+    if form.validate_on_submit():
+        _file = request.files.get('upload')
+        filename = secure_filename(_file.filename)
+        filepath = os.path.join(current_app.config.get('UPLOAD_FOLDER'), filename)
+        _file.save(filepath)
+        main(filepath, filename, None, None, None)
+        return render_template('conductor/upload_success.html', filepath=filepath)
+    else:
+        return render_template('conductor/upload_new.html', form=form)
+
+@blueprint.route('/_process_file', methods=['POST'])
```

Comment:
Since you are running the `main` function in your upload above, you can either:

1. remove this route
2. move the `main` call into this route so that the ajax works properly